py/torch_tensorrt/dynamo/_refit.py

```diff
@@ -0,0 +1,362 @@
+            [("kernel", "KERNEL"), ("bias", "BIAS")],
+        ),
+        "DECONVOLUTION": (
+            trt.IDeconvolutionLayer,
+            [("kernel", "KERNEL"), ("bias", "BIAS")],
+        ),
+        "CONSTANT": (trt.IConstantLayer, [("weights", "CONSTANT")]),
+    }
+
+    output_dtypes = infer_module_output_dtypes(
+        module,
+        inputs,
+        settings.device,
+        truncate_double=settings.truncate_double,
+    )
+
+    # Use Interpreter
+    weight_map = {}
+    interpreter = TRTInterpreter(
+        module,
+        inputs,
+        logger_level=(trt.Logger.VERBOSE if settings.debug else trt.Logger.WARNING),
+        output_dtypes=output_dtypes,
+        compilation_settings=settings,
+    )
+    interpreter._construct_trt_network_def()
+    net = interpreter.ctx.net
+    for i in range(net.num_layers):
+        layer = net[i]
+        layer_type: str = layer.type.name
+        if layer_type in MODULE_MAP:
+            layer.__class__ = MODULE_MAP[layer_type][0]
+            for weight_type, weight_name in MODULE_MAP[layer_type][1]:
+                weight = layer.__getattribute__(weight_type).copy()
+                weight_dtype = (
+                    layer.precision
+                    if layer.precision_is_set
+                    else dtype.try_from(weight.dtype).to(trt.DataType)
+                )
+                weight_map[f"{layer.name} {weight_name}"] = (
+                    weight,
+                    weight_dtype,
+                )
+
+    return weight_map
+
+
+def _refit_single_trt_engine_with_gm(
+    new_gm: torch.fx.GraphModule,
+    old_engine: trt.ICudaEngine,
+    input_list: Tuple[Any, ...],
+    settings: CompilationSettings = CompilationSettings(),
+) -> None:
+    """
+    Refit a TensorRT Engine in place
+    """
+    # Get the refitting mapping
+    mapping = construct_refit_mapping(new_gm, input_list, settings)
+    refitted = set()
+
+    trt_wt_location = trt.TensorLocation.HOST
+    refitter = trt.Refitter(old_engine, TRT_LOGGER)
+    weight_list = refitter.get_all_weights()
+
+    for layer_name in weight_list:
+        if layer_name not in mapping:
+            raise AssertionError(f"{layer_name} is not found in weight mapping")
+        # Use Numpy to create weights
+        weight, datatype = mapping[layer_name]
+        trt_wt_tensor = trt.Weights(datatype, weight.ctypes.data, weight.size)
+        refitter.set_named_weights(layer_name, trt_wt_tensor, trt_wt_location)
+        refitted.add(layer_name)
+
+    if len(refitted) != len(weight_list):
+        logger.warning("Not all weights have been refitted!!!")
+
+    if not refitter.refit_cuda_engine():
+        logger.error("Error: failed to refit new weights.")
+        exit(0)
+
+
+def refit_module_weights(
+    compiled_module: torch.fx.GraphModule | ExportedProgram,
+    new_weight_module: ExportedProgram,
+    inputs: Tuple[Any, ...],
+    verify_output: bool = True,
+) -> torch.fx.GraphModule:
+    """
```

Comment:
Add a formal docstring here telling people how to use the API and add it to `//docsrc/py_api/dynamo.rst` so that it gets rendered in the docs.